tests/sentry/issues/auto_source_code_config/test_process_event.py

```diff
@@ -92,78 +98,61 @@ def _process_and_assert_configuration_changes(
-            assert code_mapping is not None
-            assert code_mapping.repository.name == expected_repo_name
-
-            platform_config = PlatformConfig(platform)
-            dry_run = platform_config.is_dry_run_platform()
-            tags = {"dry_run": dry_run, "platform": platform}
-
-            if Repository.objects.all().count() > repositories_count:
+
+            if dry_run:
+                # If dry run, no configurations should have been created
+                assert starting_code_mappings_count == code_mappings.count()
+                assert starting_repositories_count == Repository.objects.all().count()
+
+                if expected_new_code_mapping:
+                    assert len(dry_run_code_mappings) == 1
+                    cm = dry_run_code_mappings[0]
+                    assert cm.stacktrace_root == expected_new_code_mapping["stack_root"]
+                    assert cm.source_path == expected_new_code_mapping["source_root"]
+                    assert cm.repo.name == expected_new_code_mapping["repo_name"]
+
                 mock_incr.assert_any_call(
                     key=f"{METRIC_PREFIX}.repository.created", tags=tags, sample_rate=1.0
                 )
-
-            if RepositoryProjectPathConfig.objects.all().count() > code_mappings_count:
                 mock_incr.assert_any_call(
                     key=f"{METRIC_PREFIX}.code_mapping.created", tags=tags, sample_rate=1.0
                 )
+            else:
+                assert code_mappings.count() == expected_num_code_mappings
+                if expected_new_code_mapping:
+                    assert code_mappings.count() == starting_code_mappings_count + 1
+                    code_mapping = code_mappings.filter(
+                        stack_root=expected_new_code_mapping["stack_root"],
+                        source_root=expected_new_code_mapping["source_root"],
+                    ).first()
+                    assert code_mapping is not None
+                    assert code_mapping.repository.name == expected_new_code_mapping["repo_name"]
+
+                if Repository.objects.all().count() > starting_repositories_count:
+                    mock_incr.assert_any_call(
+                        key=f"{METRIC_PREFIX}.repository.created", tags=tags, sample_rate=1.0
+                    )
+
+                if code_mappings.count() > starting_code_mappings_count:
+                    mock_incr.assert_any_call(
+                        key=f"{METRIC_PREFIX}.code_mapping.created", tags=tags, sample_rate=1.0
+                    )
+
             if expected_in_app_stack_trace_rules is not None:
                 # XXX: Grab it from the option
                 assert expected_in_app_stack_trace_rules == in_app_stack_trace_rules
 
-            # Returning this is useful to inspect hash changes
+            # Returning these to inspect the results
             return event
 
-    def _process_and_assert_no_configuration_changes(
```

Comment:
Thinking more about it. I could have made this change on a separate PR and make it a little less burdensome to review.

Let me know if you would like me to.